### PR TITLE
feat: [DEL-3944]: Remove unused parked task api's from delegate 

### PR DESCRIPTION
**Pull request by jennyjames21**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/34330)

### CODE REVIEW

**Review by raghuAtWings — APPROVED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34330)* status: :shipit: complete! ✓

File: 910-delegate-task-grpc-service/src/main/java/io/harness/task/service/impl/TaskServiceImpl.java [hunk 1]
```diff
@@ -67,85 +50,6 @@ public TaskServiceImpl(DelegateServiceGrpcAgentClient delegateServiceGrpcAgentCl
...
-      responseObserver.onNext(ExecuteParkedTaskResponse.newBuilder().setTaskId(request.getTaskId()).build());
-      responseObserver.onCompleted();
-    } catch (Exception ex) {
-      log.error("Unexpected error occurred while processing execute parked task request.", ex);
-      responseObserver.onError(io.grpc.Status.INTERNAL.withDescription(ex.getMessage()).asRuntimeException());
-    }
-  }
-
-  @Override
-  public void taskProgress(TaskProgressRequest request, StreamObserver<TaskProgressResponse> responseObserver) {
```
Comment (harshjain12) [inline 910-delegate-task-grpc-service/src/main/java/io/harness/task/service/impl/TaskServiceImpl.java:86]: This is being used in CD.
Do not remove it.

**Review by harshjain12 — CHANGES_REQUESTED** [review]
> Reviewed 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34330)* status: 1 change requests. 0 of 1 approvals obtained.

File: 910-delegate-task-grpc-service/src/main/java/io/harness/task/service/impl/TaskServiceImpl.java (hunk 1 above)
Comment (jennyjames21) [reply 910-delegate-task-grpc-service/src/main/java/io/harness/task/service/impl/TaskServiceImpl.java:86]: There are 2 taskprogress api's, one related to parked task and other sendTaskProgress which is used to send task status.i removed specific to parked task.  Why do CD team parked task? Who is contact person in CD who is aware of this flow?

**Review by harshjain12 — APPROVED** [review]
> Reviewed 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34330)* status: :shipit: complete! ✓

**Pull request merged**